dotCMS/src/integration-test/java/com/dotcms/rest/MapToContentletPopulatorTest.java

```diff
@@ -0,0 +1,210 @@
+            final Field childTypeRelationshipField = createAndSaveRelationshipField("otherSideRel",
+                    childContentType.id(), fullFieldVar);
+            final Map<String, Object> properties = new HashMap<>();
+            properties.put(Contentlet.STRUCTURE_INODE_KEY, childContentType.id());
+            properties.put(childTypeRelationshipField.variable(), contentlet.getIdentifier());
+
+            contentlet = populator.populate(contentlet, properties);
+
+            assertTrue(UtilMethods.isSet(contentlet.get(Contentlet.RELATIONSHIP_KEY)));
+
+            Map<Relationship, List<Contentlet>>  resultMap = (Map<Relationship, List<Contentlet>> ) contentlet
+                    .get(Contentlet.RELATIONSHIP_KEY);
+
+            assertEquals(1, resultMap.size());
+
+            Entry<Relationship, List<Contentlet>> result = resultMap.entrySet().iterator().next();
+
+            //validates the relationship
+            assertEquals(childContentType.inode(), result.getKey().getChildStructureInode());
+
+            //validates the contentlet
+            assertEquals(1, result.getValue().size());
+            assertEquals(contentlet.getInode(), result.getValue().get(0).getInode());
+        } finally {
+            if (UtilMethods.isSet(parentContentType) && UtilMethods.isSet(parentContentType.id())) {
+                contentTypeAPI.delete(parentContentType);
+            }
+
+            if (UtilMethods.isSet(childContentType) && UtilMethods.isSet(childContentType.id())) {
+                contentTypeAPI.delete(childContentType);
+            }
+        }
+    }
+
+    private ContentType createAndSaveSimpleContentType(final String name) throws DotSecurityException, DotDataException {
+        return contentTypeAPI.save(ContentTypeBuilder.builder(SimpleContentType.class).folder(
+                FolderAPI.SYSTEM_FOLDER).host(Host.SYSTEM_HOST).name(name)
+                .owner(user.getUserId()).build());
+    }
+
+    private Field createAndSaveRelationshipField(final String relationshipName, final String parentTypeId,
+            final String childTypeVar)
+            throws DotSecurityException, DotDataException {
+
+        final Field field = FieldBuilder.builder(RelationshipField.class).name(relationshipName)
+                .contentTypeId(parentTypeId).values( String.valueOf(RELATIONSHIP_CARDINALITY.MANY_TO_MANY.ordinal()))
+                .relationType(childTypeVar).build();
+
+        return fieldAPI.save(field, user);
+    }
+
+    private Contentlet createContentlet(final ContentType contentType) {
+        final ContentletDataGen contentletDataGen = new ContentletDataGen(contentType.id());
+        Contentlet contentlet = contentletDataGen.next();
+        contentlet.setStringProperty("widgetTitle", "New YouTube Content");
+        contentlet.setStringProperty("url", "new-youtube-content");
+        contentlet = contentletDataGen.persist(contentlet);
+        return contentlet;
```

Comment:
you can do all this in one line
return final Contentlet contentlet = new ContentletDataGen(contentType.id()).setStringProperty("widgetTitle", "New YouTube Content").setStringProperty("url", "new-youtube-content").nextPersisted();

Reply:
it is not possible

Reply:
it was updated using `setProperty` instead of `setStringProperty`